torch/csrc/jit/tensorexpr/kernel.cpp

```diff
@@ -937,9 +937,14 @@ Tensor TensorExprKernel::bindInput(const torch::jit::Value* input) {
         bufferArgs_.emplace_back(inBuffer);
         break;
       }
+      ExprHandle flat_size = 1;
+      for (size_t i = 0; i < *tt->sizes().size(); i++) {
+        flat_size = flat_size + (*tt->sizes()[i] - 1) * *tt->strides()[i];
```

Comment:
Now since we have -1 there we might need to check if `size[i]` is not 0.

Reply:
That's interesting. If one dimension has size 0, then the flat size of the tensor should be 0 right?

Reply:
> the corresopnding IR will be simplified at some later point anyway

I guess we will treat `inBuffer.load(idx)` as dead code and remove it in simplifier? This makes sense.

Reply:
Yes and yes :)